extensions/tokenizers/src/main/java/ai/djl/huggingface/translator/TextEmbeddingTranslator.java

```diff
@@ -160,18 +160,47 @@ static NDArray processEmbedding(
         NDArray inputAttentionMask = manager.create(attentionMask).toType(DataType.FLOAT32, true);
         switch (pooling) {
             case "mean":
-                return meanPool(embedding, inputAttentionMask, false);
+                embedding = meanPool(embedding, inputAttentionMask, false);
+                break;
             case "mean_sqrt_len":
-                return meanPool(embedding, inputAttentionMask, true);
+                embedding = meanPool(embedding, inputAttentionMask, true);
+                break;
             case "max":
-                return maxPool(embedding, inputAttentionMask);
+                embedding = maxPool(embedding, inputAttentionMask);
+                break;
             case "weightedmean":
-                return weightedMeanPool(embedding, inputAttentionMask);
+                embedding = weightedMeanPool(embedding, inputAttentionMask);
+                break;
             case "cls":
-                return embedding.get(0);
+                embedding = embedding.get(0);
+                break;
             default:
                 throw new AssertionError("Unexpected pooling mode: " + pooling);
         }
+
+        embedding = embedding.toDevice(Device.cpu(), false);
+        if (denseModel != null) {
+            NDArray weight = denseModel.get("linear.weight");
+            NDArray bias = denseModel.get("linear.bias");
+            embedding = embedding.getNDArrayInternal().linear(embedding, weight, bias).get(0);
+            if ("Tanh".equals(denseActivation)) {
```

Comment:
should this be `"tanh"`? do we want to compare lower case of denseActivation?

Reply:
Fixed